### ISSUE TITLE
Instrument List blank

### ISSUE THREAD

**Issue by John-Holt-Tessella** [ISISComputingGroup/IBEX#1183]
On change to instrument NDXMUONFE cannot change instrument again the list is now blank apart from my own (I assume it can not connect with PV). Stack trace is:

*2016-05-10 16:33:04.104 [Worker-0] INFO  uk.ac.stfc.isis.ibex.log.jms.JmsHandler - Successfully disconnected from JMS: tcp://NDXLARMOR:61616
*2016-05-10 16:33:04.162 [PVMgr Worker 2] INFO  uk.ac.stfc.isis.ibex.configserver.Configurations - Current config connected: true
*2016-05-10 16:33:04.162 [PVMgr Worker 3] INFO  uk.ac.stfc.isis.ibex.configserver.Configurations - Server status connected: true
*2016-05-10 16:33:04.263 [PVMgr Worker 4] INFO  uk.ac.stfc.isis.ibex.configserver.Configurations - Server status value: 
*2016-05-10 16:33:04.293 [PVMgr Worker 1] INFO  uk.ac.stfc.isis.ibex.configserver.Configurations - Current config value: TestPSUs
*2016-05-10 16:33:04.296 [PVMgr Worker 1] INFO  uk.ac.stfc.isis.ibex.configserver.Configurations - Current config {"name":"TestPSUs","description":"Example config while testing PSUs","synoptic":"TestPSUs","iocs":[{"name":"DFKPS_01","autostart":true,"restart":true,"simlevel":"none","pvsets":[],"pvs":[],"macros":[]},{"name":"GENESYS_01","autostart":true,"restart":true,"simlevel":"none","pvsets":[],"pvs":[],"macros":[]}],"blocks":[{"name":"Q10A","pv":"IN:MUONFE:GENESYS_01:1:CURR","visible":true,"local":true,"runcontrol":false,"lowlimit":0.0,"highlimit":0.0,"log_periodic":false,"log_rate":0,"log_deadband":0.0}],"groups":[{"name":"NONE","blocks":["Q10A"]}],"components":[],"history":[]}
*2016-05-10 16:33:04.364 [PVMgr Worker 4] INFO  uk.ac.stfc.isis.ibex.configserver.Configurations - Server status value: 
*2016-05-10 16:33:04.417 [PVMgr Worker 5] INFO  uk.ac.stfc.isis.ibex.configserver.Configurations - Current config value: TestPSUs
*2016-05-10 16:33:04.420 [PVMgr Worker 5] INFO  uk.ac.stfc.isis.ibex.configserver.Configurations - Current config {"name":"TestPSUs","description":"Example config while testing PSUs","synoptic":"TestPSUs","iocs":[{"name":"DFKPS_01","autostart":true,"restart":true,"simlevel":"none","pvsets":[],"pvs":[],"macros":[]},{"name":"GENESYS_01","autostart":true,"restart":true,"simlevel":"none","pvsets":[],"pvs":[],"macros":[]}],"blocks":[{"name":"Q10A","pv":"IN:MUONFE:GENESYS_01:1:CURR","visible":true,"local":true,"runcontrol":false,"lowlimit":0.0,"highlimit":0.0,"log_periodic":false,"log_rate":0,"log_deadband":0.0}],"groups":[{"name":"NONE","blocks":["Q10A"]}],"components":[],"history":[]}
*2016-05-10 16:33:06.651 [Worker-0] WARN  uk.ac.stfc.isis.ibex.log.jms.JmsHandler - Problem connecting to JMS server. Will auto-attempt reconnection.
May 10, 2016 4:33:06 PM org.csstudio.alarm.beast.ui.clientmodel.AlarmClientModel readConfiguration
SEVERE: Cannot connect to RDB
java.lang.Exception: Cannot connect to database, Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
	at org.csstudio.alarm.beast.client.AlarmConfiguration.<init>(AlarmConfiguration.java:98)
	at org.csstudio.alarm.beast.client.AlarmConfiguration.<init>(AlarmConfiguration.java:125)
	at org.csstudio.alarm.beast.ui.clientmodel.AlarmClientModel$1.<init>(AlarmClientModel.java:361)
	at org.csstudio.alarm.beast.ui.clientmodel.AlarmClientModel.readConfiguration(AlarmClientModel.java:361)
	at org.csstudio.alarm.beast.ui.clientmodel.ReadConfigJob.run(ReadConfigJob.java:49)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:54)
Caused by: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:411)
	at com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:1116)
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:344)
	at com.mysql.jdbc.ConnectionImpl.coreConnect(ConnectionImpl.java:2332)
	at com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2369)
	at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2153)
	at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:792)
	at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:47)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:411)
	at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:381)
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:305)
	at java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql.DriverManager.getConnection(Unknown Source)
	at org.csstudio.platform.utility.rdb.internal.MySQL_RDB.connect(MySQL_RDB.java:55)
	at org.csstudio.platform.utility.rdb.RDBUtil.<init>(RDBUtil.java:179)
	at org.csstudio.platform.utility.rdb.RDBUtil.connect(RDBUtil.java:143)
	at org.csstudio.alarm.beast.client.AlarmConfiguration.<init>(AlarmConfiguration.java:94)
	... 5 more
Caused by: java.net.UnknownHostException: NDXMUONFE
	at java.net.InetAddress.getAllByName0(Unknown Source)
	at java.net.InetAddress.getAllByName(Unknown Source)
	at java.net.InetAddress.getAllByName(Unknown Source)
	at com.mysql.jdbc.StandardSocketFactory.connect(StandardSocketFactory.java:247)
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:294)
	... 23 more

**Comment by IsabellaRey** [comment]
This will probably be fixed by ISISComputingGroup/IBEX#1222 but will need to check

**Comment by John-Holt-Tessella** [comment]
Also check we can read it when the list is exactly:
```
C:\Instrument\Apps\EPICS>caget -S -t CS:INSTLIST|uzhex
[{"name": "LARMOR"}, {"name": "ALF"}, {"name": "DEMO"}, {"name": "IMAT"}, {"name
": "MUONFE"}, {"name": "ZOOM"}, {"name": "IRIS"}]
C:\Instrument\Apps\EPICS>caget CS:INSTLIST
CS:INSTLIST 512 55 56 57 99 56 98 97 101 53 54 99 97 52 98 99 99 52 100 53 53 98
 50 53 50 53 48 102 50 55 49 48 99 102 50 102 53 48 102 53 50 97 97 100 53 53 49
 52 48 56 56 51 57 102 97 98 56 97 49 48 97 98 56 98 56 102 97 102 97 97 51 56 9
7 55 56 102 97 51 97 56 54 97 48 56 97 102 56 56 54 102 97 102 98 98 57 98 57 97
 50 56 97 52 53 102 57 102 98 102 98 97 50 101 57 48 98 102 50 48 99 53 54 97 97
 56 100 48 53 48 48 50 102 57 55 50 52 49 52 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
```

because this could not be read with a conversion error complaining about odd number of bytes (not logged I had to go into debug mode to find it). But when changed to:

```
C:\Instrument\Apps\EPICS>caget  CS:INSTLIST
CS:INSTLIST 512 55 56 57 99 56 98 97 101 53 54 99 97 52 98 99 99 52 100 53 53 98
 50 53 50 53 48 102 50 55 49 48 99 102 50 102 53 48 102 53 50 97 97 100 53 53 49
 52 48 56 56 51 57 102 97 98 56 97 49 48 97 98 56 98 56 102 97 102 97 97 51 56 9
7 55 56 102 97 51 97 56 54 97 48 56 97 102 56 56 54 102 97 102 98 98 57 98 57 97
 50 56 97 52 53 102 57 102 98 102 98 97 50 101 57 48 98 102 50 48 99 53 54 97 97
 53 53 56 56 48 53 48 48 53 51 56 101 50 52 51 52 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0

C:\Instrument\Apps\EPICS>caget -S -t CS:INSTLIST|uzhex
[{"name": "LARMOR"}, {"name": "ALF"}, {"name": "DEMO"}, {"name": "IMAT"}, {"name
": "MUONFE"}, {"name": "ZOOM"}, {"name": "IRIS"} ]
```

It did work.

**Comment by AdrianPotter** [comment]
I did some tests from the command line and didn't find any issue encoding or decoding either value. Eventually, I set the instrument list using the process described here:

https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Making-an-Instrument-Available-from-the-GUI

I've set it to the "non-working" value and everything seems fine. I can caget from the command line and have tried many times to launch and switch instrument with the GUI with no issue. I compared the new hexed value with the one above and they are the same so I can offer little explanation why this didn't work before
